src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerEngine.java

```diff
@@ -83,8 +86,26 @@ public class HaplotypeCallerEngine implements AssemblyRegionEvaluator {
 
     protected final OutputStreamWriter assemblyDebugOutStream;
 
-    // the genotyping engine for the isActive() determination
-    private MinimalGenotypingEngine activeRegionEvaluationGenotyperEngine = null;
+    /**
+     * List of interval file entries including regions and custom ploidy values to apply in that region.
+     */
+    protected final List<NamedFeature> ploidyRegions = new ArrayList<>();
+
+    /**
+     * List of all custom ploidies provided by user
+     */
+    private final LinkedHashSet<String> allCustomPloidies;
+
+    /**
+     * The default genotyping engine for the isActive() determination
+     */
+    private MinimalGenotypingEngine defaultActiveRegionEvaluationGenotyperEngine = null;
+
+    /**
+     * Map of user-provided ploidy values to corresponding active region genotyper. Values are checked as valid Integers during
+     * initialization, but Strings are used as keys to avoid parsing repeatedly during runtime.
```

Comment:
Why not do all the parsing upon initialization.  As I see it, you read in the intervals file once, parsing the string ploidies as ints, then store them in some kind of object that holds an interval and an int.  Maybe `NamedFeature` can't do this.  We already have a GATK class called `SimpleCount` that does.  You could use it and perhaps change its name if you like.

Reply:
This is great. I switched to using `SimpleCount`, and added a constructor to parse a `NamedFeature` into one.